### PR TITLE
feat: quick calculations

### PR DESCRIPTION
**Pull request by rephus**
Closes:   #9349

### Description:

Screenshot from 2024-03-19 16-41-50

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

**Review by TuringLovesDeathMetal — CHANGES_REQUESTED** [review]
woweeee! so exciting!

This looks really great :) I think the only thing that isn't working as expected for me is the ordering in the table calculation. 

I think the ordering in the `ORDER BY` bit of the table calculation should match whatever's in my results table ordering. 

e.g. here:

image


I'd expect the `order by` to be `payments_method, order_date_month DESC`

This is what it actually is: 

image

So, two things:

1. The `desc` is applied when I've ordered a column descending (e.g. date month here)
2. There is only ordering applied to the columns that have ordering in the table (e.g. here, there is no ordering applied to `total revenue`, so it doesn't get ordering in the table calculation).

**Comment by TuringLovesDeathMetal** [comment]
sorry, one more thing: have we built analytics events for this? (I think that a frontend event might be the best here 👀, but not sure if it's possible to make it a backend event ).

aaaand docs - not sure if you just wanted a review on functionality first! Then were going to do them separately :) but let me know if you need help!

**Comment by rephus** [comment]
Hey @TuringLovesDeathMetal thanks for the review

I fixed the sorting, one thing to notice, if you change the sorting , the table calculation will not get automatically updated ... 
image

> have we built analytics events for this
There is already a `create_quick_table_calculation_button.click`  there is no backend for this

> aaaand docs
Also done https://github.com/lightdash/lightdash/pull/9452/files#diff-aefcc74f448e37d7af568b590923f62fd8f8ca9e441b33a16cdbcc42aab8c1ffR40

File: docs/docs/guides/table-calculations/adding-table-calculations.mdx [hunk 1]
```diff
@@ -32,11 +33,26 @@ Watch out: table calculations can be easier to create than regular metrics/dimen
...
 Table calculations can only be built using the dimensions + metrics that you've included in your results table. So, to create a table calculation, first, you need to add the dimensions and/or metrics for your table calculation to your results table.
 
-#### 2. To add a table calculation just click on `+ Table calculation` in the `Results` tab
+### Using Quick Calculations
+
+Quick calculations are shortcuts to the most common table calculations, these are only available to metrics in the contextual menu in the results table.
+
+<img
+  src={QuickTableCalculation}
+  width="880"
```
Comment (TuringLovesDeathMetal) [inline docs/docs/guides/table-calculations/adding-table-calculations.mdx:46]:
```suggestion
  width="352"
```

File: docs/docs/guides/table-calculations/adding-table-calculations.mdx [hunk 2]
```diff
@@ -32,11 +33,26 @@ Watch out: table calculations can be easier to create than regular metrics/dimen
...
+  height="557"
```
Comment (TuringLovesDeathMetal) [inline docs/docs/guides/table-calculations/adding-table-calculations.mdx:47]:
```suggestion
  height="223"
```

File: docs/docs/guides/table-calculations/adding-table-calculations.mdx [hunk 3]
```diff
@@ -32,11 +33,26 @@ Watch out: table calculations can be easier to create than regular metrics/dimen
...
+  style={{ display: 'block', margin: '0 auto 20px auto' }}
+/>
+
```
Comment (TuringLovesDeathMetal) [inline docs/docs/guides/table-calculations/adding-table-calculations.mdx:50]:
```suggestion
To learn more about what these calculations are doing, [check out the docs here](https://docs.lightdash.com/guides/table-calculations/sql-templates).
```

**Review by TuringLovesDeathMetal — CHANGES_REQUESTED** [review]
Niiiice! 

Ya, I think the image is definitely a bit big, I've added some suggested size for it (normally, I just use 40% of the image size and it's pretty good?). 

I've also linked out to the table calculation templates which explain each calculation. 

And yep! That makes sense for the sorting - playing with it definitely makes me feel like this ist he first step to sql-less table calculations 👀

**Comment by TuringLovesDeathMetal** [comment]
@rephus sorry, my markdown was obviously wrong, I don't think the link rendered properly

image

**Comment by rephus** [comment]
weird, I copied the same url we have in the doc and still it doesn't work 

image

**Comment by rephus** [comment]
ok, this worked. 
image

**Review by TuringLovesDeathMetal — APPROVED** [review]
amazing! looks great 🚀

**Pull request merged**